otherlibs/stdune/src/string.mli

```diff
@@ -52,7 +52,15 @@ val extract_blank_separated_words : t -> t list
 val lsplit2 : t -> on:char -> (t * t) option
 val lsplit2_exn : t -> on:char -> t * t
 val rsplit2 : t -> on:char -> (t * t) option
+
+(** [split t ~on] returns the list of non-overlapping substrings of
+    [t] between each occurence of [on]. If [t] begins or ends with [on]
+    then an empty string will be present on the "far" side of [on] in the
+    output. If [on] does not appear in [t] then the result is a list
+    containing [t] (even if [t] is the empty string). This function never
+    returns the empty list. *)
```

Comment:
I think the idea here is that joining a string that was split would return the original string, this it needs to have empty strings on the far ends, so the original locations of the separators can be recovered.

Maybe worth mentioning.

Reply:
Makes sense. Will update. To be clear I think the way `split` works is very natural. Just wanted to make the doc explicitly state how it behaves around edge cases.